### ISSUE TITLE
Error when compiling with -DOpenSSL_ENABLE=OFF

### ISSUE THREAD

**Issue by sergneo** [fireice-uk/xmr-stak#461]
cmake -DOpenSSL_ENABLE=OFF -G "Visual Studio 2017 15 Win64" -T v140,host=x64 ..
cmake --build . --config Release --target install
Warnings: 80
Error: 32

Windows 7 x64 sp1, Visual Studio 2017, CUDA 8, Cmake 3.10.0

**Comment by psychocrypt** [comment]
Please post the full output of the compile in an empty build folder.

**Comment by Cleeez** [comment]
I had the same issue, only could get it to compile with openSSL on

**Comment by sergneo** [comment]
```
C:\xmr-stak-2.1.0\build>cmake -DOpenSSL_ENABLE=OFF -DCUDA_ENABLE=OFF -DOpenCL_EN
ABLE=OFF -G "Visual Studio 15 2017 Win64" -T v140,host=x64 ..
-- The C compiler identification is MSVC 19.0.24225.1
-- The CXX compiler identification is MSVC 19.0.24225.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studi
o 14.0/VC/bin/amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studi
o 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Set miner currency to 'monero' and 'aeon'
-- Configuring done
-- Generating done
-- Build files have been written to: C:/xmr-stak-2.1.0/build
```

**Comment by sergneo** [comment]
And this compilation without a key.

```
C:\xmr-stak-2.1.0\build>cmake -DOpenCL_ENABLE=OFF -DCUDA_ENABLE=OFF -G "Visual S
tudio 15 2017 Win64" -T v140,host=x64 ..
-- Set miner currency to 'monero' and 'aeon'
-- Configuring done
-- Generating done
-- Build files have been written to: C:/xmr-stak-2.1.0/build

C:\xmr-stak-2.1.0\build>cmake -DOpenCL_ENABLE=OFF -DCUDA_ENABLE=OFF -G "Visual S
tudio 15 2017 Win64" -T v140,host=x64 ..
-- The C compiler identification is MSVC 19.0.24225.1
-- The CXX compiler identification is MSVC 19.0.24225.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studi
o 14.0/VC/bin/amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studi
o 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Set miner currency to 'monero' and 'aeon'
-- Found OpenSSL: C:/xmr-stak-dep/openssl/lib/libeay32.lib (found version "1.0.2
l")
-- Configuring done
-- Generating done
-- Build files have been written to: C:/xmr-stak-2.1.0/build
```

**Comment by NmxMilk** [comment]
error message indicates missing windows includes 
```C
#include <ShellApi.h>
```
I guess this is because openssl lib includes at some stage <openssl/rand.h> which includes <windows.h>

**Comment by maztheman** [comment]
If you include shellapi.h you get a bunch of different errors.

**Comment by sergneo** [comment]
Well, this problem is solved? In version 2.0.0 of such errors did not occur.
